vod/media_set_parser.c

```diff
@@ -2485,9 +2486,21 @@ media_set_parse_json(
 	{
 		result->presentation_end = TRUE;
 	}
-	else if (params[MEDIA_SET_PARAM_PLAYLIST_TYPE]->v.str.len == playlist_type_live.len &&
-		vod_strncasecmp(params[MEDIA_SET_PARAM_PLAYLIST_TYPE]->v.str.data, playlist_type_live.data, playlist_type_live.len) == 0)
+	else
 	{
+		if (params[MEDIA_SET_PARAM_PLAYLIST_TYPE] != NULL &&
```

Comment:
no need to check params[MEDIA_SET_PARAM_PLAYLIST_TYPE] != NULL here, if we entered the 'else', it's surely not NULL